connection_test.go

```diff
@@ -1322,3 +1322,28 @@ func TestInvalidTransportHeaders(t *testing.T) {
 		})
 	}
 }
+
+func TestCustomDialer(t *testing.T) {
+	sopts := testutils.NewOpts().NoRelay()
```

Comment:
add a comment for why are doing `NoRelay`

it would be great if we didn't have that option and made the test work for relays as well

Reply:
I copied most of the test from "TestPeerStatusChangeClient".  So inherited setting NoRelay from there.  Able to get the test to pass with NoRelay to removing it.